realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -323,22 +337,86 @@ public E remove(int location) {
     }
 
     /**
-     * Removes all elements from this list and delete them from the corresponding Realm. This method can be called on a
-     * non-managed {@link RealmList} if all of the RealmObjects in the list are managed by Realm.
+     * Removes one instance of the specified object from this {@code Collection} if one
+     * is contained . This implementation iterates over this
+     * {@code Collection} and tests for each element {@code e} returned by the iterator,
+     * whether {@code e} is equal to the given object. If {@code object != null}
+     * then this test is performed using {@code object.equals(e)}, otherwise
+     * using {@code object == null}. If an element equal to the given object is
+     * found, then the {@code remove} method is called on the iterator and
+     * {@code true} is returned, {@code false} otherwise. If the iterator does
+     * not support removing elements, an {@code UnsupportedOperationException}
+     * is thrown.
      *
-     * @throws IllegalStateException if the Realm instance has been closed, the parent object has been removed, the
-     * method is called in a wrong thread or any RealmObject in the list is not managed by Realm.
-     * @see #clear()
+     * @param object the object to remove.
+     * @return {@code true} if this {@code Collection} is modified, {@code false} otherwise.
+     * @throws ClassCastException if the object passed is not of the correct type.
+     * @throws NullPointerException  if {@code object} is {@code null}.
+     */
+    @Override
+    public boolean remove(Object object) {
+        if (managedMode && !realm.isInTransaction()) {
+            throw new IllegalStateException("Objects can only be removed from inside a write transaction");
+        }
+        return super.remove(object);
+    }
+
+    /**
+     * Removes all occurrences in this {@code Collection} of each object in the
+     * specified {@code Collection}. After this method returns none of the
+     * elements in the passed {@code Collection} can be found in this {@code Collection}
+     * anymore.
+     * <p>
+     * This implementation iterates over this {@code Collection} and tests for each
+     * element {@code e} returned by the iterator, whether it is contained in
+     * the specified {@code Collection}. If this test is positive, then the {@code
+     * remove} method is called on the iterator.
+     *
+     * @param collection the collection of objects to remove.
+     * @return {@code true} if this {@code Collection} is modified, {@code false} otherwise.
+     * @throws ClassCastException if one or more elements of {@code collection} isn't of the correct type.
+     * @throws NullPointerException if {@code collection} contains at least one {@code null} element.
+     * @throws NullPointerException if {@code collection} is {@code null}.
+     */
+    @Override
+    public boolean removeAll(Collection<?> collection) {
```

Comment:
As long as we don't support inheritance, it should be impossible to call this method with the wrong types (since `RealmObject` is an abstract class) ?